tests/test_tools.py

```diff
@@ -27,5 +32,114 @@ def test_ClientRedirectServer(self):
         self.assertEqual(httpd.query_params.get('code'), code)
 
 
+class _Namespace(object):
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+
+class TestRunFlow(unittest2.TestCase):
+    @mock.patch.object(sys, 'argv',
+        [sys.executable, '--noauth_local_webserver'])
+    @mock.patch('oauth2client.tools.logging')
+    @mock.patch('oauth2client.tools.input')
+    def test_run_flow_no_webserver(self, input_mock, logging_mock):
+        flow = mock.Mock()
+        storage = mock.Mock()
+        credentials = mock.Mock()
```

Comment:
MagicMock also mocks the magic (`__x__`) methods, typically you don't need to use it unless you actually need those methods.